stdlib/public/SwiftShims/RefCount.h

```diff
@@ -239,6 +276,45 @@ class StrongRefCount {
   }
 
   template <bool ClearPinnedFlag>
+  bool doDecrementShouldDeallocateNonAtomic() {
+    // If we're being asked to clear the pinned flag, we can assume
+    // it's already set.
+    constexpr uint32_t quantum =
+      (ClearPinnedFlag ? RC_ONE + RC_PINNED_FLAG : RC_ONE);
+    uint32_t val = __atomic_load_n(&refCount, __ATOMIC_RELAXED);
+    val -= quantum;
+    __atomic_store_n(&refCount, val, __ATOMIC_RELAXED);
```

Comment:
I don't think this intermediate store and later assertion really helps us here, since without a barrier we aren't guaranteed to interact with updates from other threads anyway.